mono/utils/mono-proclib.c

```diff
@@ -657,6 +659,40 @@ get_pid_status_item (int pid, const char *item, MonoProcessError *error, int mul
 		ret = th_count;
 	} else if (strcmp (item, "VmSwap") == 0)
 		ret = t_info.compressed;
+#else /* Fallback code for macOS < 10.9 */
+	struct task_basic_info t_info;
+	mach_msg_type_number_t th_count = TASK_BASIC_INFO_COUNT;
+	kern_return_t mach_ret;
+
+	if (pid == getpid ()) {
+		/* task_for_pid () doesn't work on ios, even for the current process */
```

Comment:
could be simplified unless you also want to target legacy iOS

Reply:
iOS is not my concern, so how better to rewrite this chunk?